src/number_theory/pell.lean

```diff
@@ -38,6 +40,23 @@ Pell's equation
 * Connect solutions to the continued fraction expansion of `√d`.
 -/
 
+namespace zsqrtd
+
+-- Note: We put this here to avoid having to import `algebra.star.unitary`
+--       into `number_theory.zsqrtd.basic` just for this one result.
+
+/-- An element of `ℤ√d` has norm one (i.e., `a.re^2 - d*a.im^2 = 1`) if and only if
+it is contained in the submonoid of unitary elements. -/
+lemma norm_eq_one_iff_mem_unitary {d : ℤ} {a : zsqrtd d} :
+  a.re ^ 2 - d * a.im ^ 2 = 1 ↔ a ∈ unitary (zsqrtd d) :=
+begin
+  rw [unitary.mem_iff, ← norm_eq_mul_conj, mul_comm _ a, ← norm_eq_mul_conj, and_self, norm_def,
```

Comment:
`unitary.mem_iff_star_mul_self` or `unitary.mem_iff_self_mul_star` probably shortens the prof here.

Reply:
Ah, OK; I had overlooked the commutative variants!